### PR TITLE
fix(tree): update parent item in values setter

### PR DESCRIPTION
**Pull request by Sarin-Udompanish**
## Description
When set tree values via `values`, parent node doesn't show correct selected state because tree using setter from List that doesn't updating the parent node.

## Fixes # (issue)
Parent node displays wrong state when set values programmatically ELF-2020 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/tree/elements/tree.ts [hunk 1]
```diff
@@ -425,8 +425,34 @@ export class Tree<T extends TreeDataItem = TreeDataItem> extends List<T> {
       return this.composer.getItemPropertyValue(item, 'value') as string || '';
     });
   }
-  public set values (value: string[]) {
-    super.values = value;
+  public set values (values: string[]) {
+    if (!Array.isArray(values)) {
```
Comment (Nantawat-Poothong) [inline packages/elements/src/tree/elements/tree.ts:429]: Why need to check Array.isArray?  values type only accept array string already, isn't it?

File: packages/elements/src/tree/elements/tree.ts [hunk 2]
```diff
@@ -425,8 +425,34 @@ export class Tree<T extends TreeDataItem = TreeDataItem> extends List<T> {
...
+      valueFormatWarning.show();
+      this.values = [];
+    }
+    else {
+      // Clone value arrays
+      const newValue = values.slice();
```
Comment (Nantawat-Poothong) [inline packages/elements/src/tree/elements/tree.ts]: Why need to clone value array? you can use `const newComparison = values.sort().toString()` directly, Isn't it?

File: packages/elements/src/tree/elements/tree.ts (hunk 1 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/tree/elements/tree.ts:429]: It not related to the type definition. User may set values = 'some string' and we should notify an error message.

File: packages/elements/src/tree/elements/tree.ts (hunk 2 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/tree/elements/tree.ts]: You should not sort the real values.

Comment (Nantawat-Poothong) [reply packages/elements/src/tree/elements/tree.ts]: How about `const newComparison = [...values].sort().toString()` I'm not sure, what do you think?

Comment (Sarin-Udompanish) [reply packages/elements/src/tree/elements/tree.ts]: fixed

**Review by Nantawat-Poothong — DISMISSED** [review]

**Review by Nantawat-Poothong — APPROVED** [review]

**Pull request merged**